pynicotine/logfacility.py

```diff
@@ -220,6 +222,38 @@ def update_folder_paths(self):
     def open_log(self, folder_path, basename):
         self._log_file_operation(folder_path, basename, self.open_log_callback)
 
+    def _get_log_lines(self, file_handle, num_lines):
+
+        file_handle.seek(0, os.SEEK_END)
+
+        blocks = deque()
+        block_number = 1
+        file_size = file_handle.tell()
```

Comment:
Suggest storing the starting `read_offset` in the `LogFile` object, because that would then allow the log history to be prepended at any time after application startup (for example when switching to the relevant chat tab), thereby improving startup time even further.

Reply:
We can do this in your PR.